source/core/GenericFunctions.cpp

```diff
@@ -1977,18 +1985,18 @@ VIREO_FUNCTION_SIGNATURET(VectorVectorBinaryOp, AggregateBinOpInstruction)
         return _NextInstruction();
     }
 
-    while (destArray1IterPtr != NULL) {
+    while (isInputArraysDimensionsSame ? (destArray1IterPtr != destArray1EndIterPos) : (destArray1IterPtr != NULL)) {
         snippet->_p0 = srcArray1IterPtr;
         snippet->_p1 = srcArray2IterPtr;
         snippet->_p2 = destArray1IterPtr;
         if (destArray2)
            snippet->_p3 = destArray2IterPtr;
         _PROGMEM_PTR(snippet, _function)(snippet);
-        srcArray1IterPtr = (AQBlock1 *)srcArray1Iter.Next();
-        srcArray2IterPtr = (AQBlock1 *)srcArray2Iter.Next();
-        destArray1IterPtr = (AQBlock1 *)destArray1Iter.Next();
+        srcArray1IterPtr = isInputArraysDimensionsSame ? (srcArray1IterPtr + elementSize1) : (AQBlock1 *)srcArray1Iter.Next();
+        srcArray2IterPtr = isInputArraysDimensionsSame ? (srcArray2IterPtr + elementSize2) : (AQBlock1 *)srcArray2Iter.Next();
+        destArray1IterPtr = isInputArraysDimensionsSame ? (destArray1IterPtr + elementSizeDest) : (AQBlock1 *)destArray1Iter.Next();
         if (destArray2)
-           destArray2IterPtr = (AQBlock1 *)destArray2Iter.Next();
+           destArray2IterPtr = isInputArraysDimensionsSame ? (destArray2IterPtr + elementSizeDest) : (AQBlock1 *)destArray2Iter.Next();
```

Comment:
There's still a performance issue with testing isInputArraysDimensionsSame 4 different times for every element in the loop.
Better to test it once, and split out the 4 assignments into an if and else branch, avoiding ?: operator.
Actually, better to duplicate the whole while loop and test outside it.  This is the driver loop for all array/array binops, so performance is important.